_ml-commons-plugin/api/memory-apis/get-memory.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: default
+title: Get memory
+parent: Memory APIs
+grand_parent: ML Commons APIs
+nav_order: 20
+---
+
+# Get a memory
+**Introduced 2.12**
+{: .label .label-purple }
+
+Retrieves a conversational memory for [conversational search]({{site.url}}{{site.baseurl}}/search-plugins/conversational-search/). 
+
+To retrieve memory information, you can:
+
+- [Get a memory by ID](#get-a-memory-by-id)
+- [Search for a memory](#search-for-a-memory)
+- [Get all memories](#get-all-memories)
+
+To retrieve message information for a memory, you can:
+
+- [Get all messages within a memory]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/memory-apis/get-message#get-all-messages-within-a-memory) 
+- [Search for messages within a memory]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/memory-apis/get-message#search-for-messages-within-a-memory)
+
+When the Security plugin is enabled, all memories exist in a `private` security mode. Only the user who created a memory can interact with that memory and its messages.
+{: .important}
+
+## Get a memory by ID
+
+You can retrieve memory information using the `memory_id`. The response includes all messages within the memory.
+
+### Path and HTTP methods
+
+```json
+GET /_plugins/_ml/memory/<memory_id>
+```
+### Path parameters
+
+The following table lists the available path parameters.
+
+Parameter | Data type | Description
+:--- | :--- | :---
+`memory_id` | String | The ID of the memory to retrieve.
+
+#### Example request
+
+```json
+GET /_plugins/_ml/memory/N8AE1osB0jLkkocYjz7D
+```
+{% include copy-curl.html %}
+
+#### Example response
+
+```json
+{
+  "memory_id": "gW8Aa40BfUsSoeNTvOKI",
+  "create_time": "2024-02-02T18:07:06.887061463Z",
+  "updated_time": "2024-02-02T19:01:32.121444968Z",
+  "name": "Conversation for a RAG pipeline, updated",
+  "user": "admin"
+}
+```
+
+## Search for a memory
+
+Use this command to search for memories.
+
+### Path and HTTP methods
+
+```json
+GET /_plugins/_ml/memory/_search
+POST /_plugins/_ml/memory/_search
+```
+
+#### Example request: Searching for all memories
+
+```json
+POST /_plugins/_ml/memory/_search
+{
+  "query": {
+    "match_all": {}
+  },
+  "size": 1000
+}
+```
+{% include copy-curl.html %}
+
+#### Example request: Searching for memory by name
+
+```json
+POST /_plugins/_ml/memory/_search
+{
+  "query": {
+    "term": {
+      "name": {
+        "value": "conversation"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+#### Example response
+
+```json
+{
+  "conversations": [
```

Comment:
This is the response of List_Memory, not the search memory. Also, the list memory should have "memories" rather than "conversations".

Reply:
This is because these two APIs are combined on one page. I will separate the search and get APIs for all applicable APIs.